pylabrobot/resources/carrier.py

```diff
@@ -276,7 +287,31 @@ def __init__(
     category="tube_carrier",
     model: Optional[str] = None):
     super().__init__(name, size_x, size_y, size_z,
-      sites,category=category, model=model)
+      sites,category=category, model=model)\
+    
+    def assign_child_resource(self, resource, location, reassign=False):
+      if not isinstance(resource, CarrierSite):
+          print(f"Type of resource: {type(resource)}")
+          print(f"Resource details: {resource}")
+          raise TypeError(f"Invalid resource {resource}")
+      self.sites.append(resource)
+      super().assign_child_resource(resource, location=location)
```

Comment:
`Carrier.assign_child_resource` has this already. I don't immediately see what's different here, but if I missed something, could you fix it in `Carrier` so it's fixed for all of them?